lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb

```diff
@@ -163,7 +163,8 @@ def create_module_map(native_target)
               linked_path = target.module_map_path
               if path != linked_path
                 linked_path.dirname.mkpath
-                FileUtils.ln_sf(path, linked_path)
+                source = path.relative_path_from(linked_path.dirname)
+                FileUtils.ln_sf(source, linked_path)
```

Comment:
please also check `pod_target_installer.rb` which overrides this method and also does sym linking.